packages/oas-to-har/src/index.js

```diff
@@ -187,66 +217,119 @@ module.exports = (
-          // We have to clone the body object, otherwise the form
-          // will attempt to re-render with an object, which will
-          // cause it to error!
-          let cloned = removeUndefinedObjects(JSON.parse(JSON.stringify(formData.body)));
-          jsonTypes.forEach(prop => {
-            // Attempt to JSON parse each of the json properties
-            // if this errors, it'll just get caught and stringify it normally
-            cloned[prop] = JSON.parse(cloned[prop]);
+        const cleanBody = removeUndefinedObjects(JSON.parse(JSON.stringify(formData.body)));
+        if (cleanBody !== undefined) {
+          Object.keys(cleanBody).forEach(name => {
+            // We neither have an easy way to transform `name` into `name[]` to signify that it's an array payload (and
+            // for all we know it might be an array of objects!), but also at the same time the HAR spec doesn't give
+            // any guidance for these kinds of cases so instead we're just falling back to stringifying the content
+            // instead of potentially including `fileName` properties.
+            if (Array.isArray(cleanBody[name])) {
+              har.postData.params.push({
+                name,
+                value: JSON.stringify(cleanBody[name]),
+              });
+
+              return;
+            }
+
+            const data = {
+              name,
+              value: String(cleanBody[name]),
+            };
+
+            // When we want to decode the data URL for the purpose of creating a code snippet that uses the file and not
+            // the data URL we should exclude the data URL/blob fromt he parameters data.
+            if (opts.decodeDataUrl && binaryTypes.includes(name)) {
+              const decoded = dataUriToBuffer(data.value);
+              data.fileName = getNameFromDataUrlType(decoded.typeFull);
+              data.contentType = decoded.type;
+            }
+
+            har.postData.params.push(data);
           });
-
-          if (typeof cloned.RAW_BODY !== 'undefined') {
-            cloned = cloned.RAW_BODY;
+        }
+      } else if (
+        matchesMimeType(['application/json', 'application/x-json', 'text/json', 'text/x-json', '+json'], contentType)
+      ) {
+        har.postData.mimeType = contentType;
+
+        try {
+          // Find all `{ type: string, format: json }` properties in the schema because we need to manually JSON.parse
+          // them before submit, otherwise they'll be escaped instead of actual objects.
+          const jsonTypes = Object.keys(schema.schema.properties).filter(
+            key => schema.schema.properties[key].format === 'json'
+          );
+
+          if (jsonTypes.length) {
+            // We have to clone the body object, otherwise the form will attempt to re-render with an object, which will
+            // cause it to error!
+            try {
+              let cloned = removeUndefinedObjects(JSON.parse(JSON.stringify(formData.body)));
```

Comment:
Can we re-use `cleanBody` from above?

Reply:
Not easily. I'd need to wrap 100+ lines or so within a try-catch and this code is a lot easier to parse without that.

Reply:
Eh, maybe actually it'll look okay.